sample-apps/go-sample-app/collection/http_traces.go

```diff
@@ -0,0 +1,104 @@
+package collection
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+	"github.com/bitly/go-simplejson"
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/trace"
+)
+
+// Contains all of the endpoint logic.
+var tracer = otel.Tracer("sample-app")
+
+// AwsSdkCall mocks a request to s3. ListBuckets are nil so no credentials are needed.
+// Generates an Xray Trace ID.
+func (rqmc *requestBasedMetricCollector) AwsSdkCall(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+
+	s, err := session.NewSession(&aws.Config{
+		Region: aws.String("us-west-2")},
+	)
+
+	svc := s3.New(s)
+	svc.ListBuckets(nil) // nil or else would need real aws credentials
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	_, span := tracer.Start(
+		context.Background(),
+		"AWS-SDK-CALL-TRACE",
+	)
+	defer span.End()
+
+	// Request based metrics provided by rqmc
+	rqmc.AddApiRequest()
+	rqmc.UpdateTotalBytesSent()
+	rqmc.UpdateLatencyTime()
+
+	xrayTraceID := getXrayTraceID(span)
+	json := simplejson.New()
+	json.Set("traceId", xrayTraceID)
+	payload, _ := json.MarshalJSON()
+
+	w.Write(payload)
+}
+
+// OutgoingSampleApp makes a request to another Sampleapp and generates an Xray Trace ID.
+func (rqmc *requestBasedMetricCollector) OutgoingSampleApp(w http.ResponseWriter, r *http.Request) {
+	// TODO
+
+	// Request based metrics provided by rqmc
+	rqmc.AddApiRequest()
+	rqmc.UpdateTotalBytesSent()
+	rqmc.UpdateLatencyTime()
+
+}
+
+// OutgoingHttpCall makes an HTTP GET request to https://aws.amazon.com and generates an Xray Trace ID.
+func (rqmc *requestBasedMetricCollector) OutgoingHttpCall(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	client := http.Client{Transport: otelhttp.NewTransport(http.DefaultTransport)}
```

Comment:
Don't construct a new `http.Client` for every request.  Create one once in the initializer and then re-use it.

Reply:
Would this require a refactor of the endpoint functions ? How could i pass the client into these functions

Reply:
NVM, i fixed this by making the endpoints take a wrapper function and passing the client